staging/kos/cmd/controllers/connectionagent/main.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	kosclientset "k8s.io/examples/staging/kos/pkg/client/clientset/versioned"
+	cactlr "k8s.io/examples/staging/kos/pkg/controllers/connectionagent"
+	netfabricfactory "k8s.io/examples/staging/kos/pkg/networkfabric/factory"
+)
+
+const (
+	nodeNameEnv      = "NODE_NAME"
+	hostIPEnv        = "HOST_IP"
+	netFabricTypeEnv = "NET_FABRIC_TYPE"
+)
+
+func main() {
+	var kubeconfigFilename string
+	var workers int
+	var clientQPS, clientBurst int
+	flag.StringVar(&kubeconfigFilename, "kubeconfig", "", "kubeconfig filename")
+	flag.IntVar(&workers, "workers", 2, "number of worker threads")
+	flag.IntVar(&clientQPS, "qps", 100, "limit on rate of calls to api-server")
+	flag.IntVar(&clientBurst, "burst", 200, "allowance for transient burst of calls to api-server")
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+
+	glog.Infof("ConnectionAgent controller start, kubeconfig=%q, workers=%d, QPS=%d, burst=%d\n",
+		kubeconfigFilename,
+		workers,
+		clientQPS,
+		clientBurst)
+
+	if len(os.Getenv("GOMAXPROCS")) == 0 {
+		runtime.GOMAXPROCS(runtime.NumCPU())
+	}
+
+	clientCfg, err := clientcmd.BuildConfigFromFlags("", kubeconfigFilename)
+	if err != nil {
+		glog.Errorf("Failed to build client config for kubeconfig=%q: %s\n", kubeconfigFilename, err.Error())
+		os.Exit(2)
+	}
+	clientCfg.QPS = float32(clientQPS)
+	clientCfg.Burst = clientBurst
+
+	kcs, err := kosclientset.NewForConfig(clientCfg)
+	if err != nil {
+		glog.Errorf("Failed to build KOS clientset for kubeconfig=%q: %s\n", kubeconfigFilename, err.Error())
+		os.Exit(3)
+	}
+
+	localNodeName, localNodeEnvSet := os.LookupEnv(nodeNameEnv)
```

Comment:
How about using the hostname as the default for node name?

Reply:
Is the k8s node name always the hostname, or could it be different? If they are always the same I agree with your suggestion.

Reply:
Implemented the suggestion in d0dce6b.